Description: Fix documentation building
This cherry-pick 33c1c9390331b3bacd3791b537b6a1147715925c from upstream to
fix documentation building.

Author: Julien Danjou <acid@debian.org>

---
Origin: upstream, 33c1c9390331b3bacd3791b537b6a1147715925c
Bug: #889287
Forwarded: not-needed

--- keystone-2012.1~e1.orig/doc/generate_autodoc_index.py
+++ keystone-2012.1~e1/doc/generate_autodoc_index.py
@@ -13,6 +13,8 @@ base_dir = os.path.dirname(os.path.abspa
 RSTDIR=os.path.join(base_dir, "source", "sourcecode")
 SOURCEDIR=os.path.join(base_dir, "..")
 
+# Exclude these modules front the autodoc results
+EXCLUDE_MODULES = ['keystone.backends.sqlalchemy.migrate_repo.manage']
 
 def find_autodoc_modules(module_name, sourcedir):
     """returns a list of modules in the SOURCE directory"""
@@ -32,8 +34,9 @@ def find_autodoc_modules(module_name, so
                 if not (base == "__init__"):
                     elements.append(base)
                 result = (".".join(elements))
-                print result
-                modlist.append(result)
+                if result not in EXCLUDE_MODULES:
+                    print result
+                    modlist.append(result)
     return modlist
 
 if not(os.path.exists(RSTDIR)):
